include/cetl/pmr/function.hpp

```diff
@@ -0,0 +1,321 @@
+/// @copyright
+/// Copyright (C) OpenCyphal Development Team  <opencyphal.org>
+/// Copyright Amazon.com Inc. or its affiliates.
+/// SPDX-License-Identifier: MIT
+
+#ifndef CETL_PMR_FUNCTION_H_INCLUDED
+#define CETL_PMR_FUNCTION_H_INCLUDED
+
+#ifndef CETL_H_ERASE
+#    include "cetl/cetl.hpp"
+#endif
+
+#include "cetl/pf17/cetlpf.hpp"
+#include "cetl/unbounded_variant.hpp"
+
+#include <cstddef>
+#include <exception>
+#include <functional>
+
+namespace cetl
+{
+namespace pmr
+{
+
```

Comment:
This PR is more concentrated about `unbounded_variant` and adding PMR (and `valueless_by_exception`!).
That is why `function` and `InterfacePtr` not **YET** properly documented, and don't have yet proper error handling/indication - these will be addressed in the next PR (please).